File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.cs
using System;

public class Solution 
{
    public int[,] solution(int[,] arr) 
    {
        int arrRow = arr.GetLength(0);
        int arrCol = arr.GetLength(1);
                
        int row = 0;
        int col = 0;
        if(arrRow > arrCol)
        {
            row = arrRow;
            col = row;
        }
        else if(arrRow < arrCol)
        {
            col = arrCol;
            row = col;
        }
        else 
        {
            row = arrRow;
            col = arrCol;
        }
        
        int[,] arrSquare = new int[row,col];
        
        for(int i=0; i<arrRow; i++)
        {
            for(int j=0; j<arrCol; j++)
            {
                arrSquare[i,j] = arr[i,j];
            }
        }
        return arrSquare;
    }
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr) 
{
    int arrRow = arr.size();
    int arrCol = arr[0].size();
    
    int n = 0;    
    if(arrRow > arrCol)
    {
        n = arrRow;
    }
    else
    {
        n = arrCol;
    }
    
    vector<vector<int>> vec(n, vector<int> (n, 0));
    for(int i=0; i<arr.size(); i++)
    {
        vector<int> a = arr[i];
        vector<int> v = vec[i];
        
        copy(a.begin(), a.end(), v.begin());
        vec[i] = v;
    }
    return vec;
}